From 84fbdd0f4ef58f75b7ff1682a399e78a1238c0bf Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Wed, 7 Jun 2017 18:00:08 +0200
Subject: [PATCH 04/45] glmemory: Fix n_wrapped_pointers usage

gst_gl_memory_setup_buffer() was not properly using the number
of pointers to wrapped. This also fixes the validation, as we
only support 1 wrapper per view, or num_planes * views wrapper.

https://bugzilla.gnome.org/show_bug.cgi?id=783521
(cherry picked from commit 0ac646f7f4aa43399084dd63143cd47030885f5f)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 gst-libs/gst/gl/gstglmemory.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglmemory.c b/gst-libs/gst/gl/gstglmemory.c
index 30d5f5e475..c65237334f 100644
--- a/gst-libs/gst/gl/gstglmemory.c
+++ b/gst-libs/gst/gl/gstglmemory.c
@@ -1458,8 +1458,13 @@ gst_gl_memory_setup_buffer (GstGLMemoryAllocator * allocator,
   else
     views = 1;
 
+  if (n_wrapped_pointers == views)
+    n_mem = 1;
+
+  /* Sanity check for the code below; there should be as many pointers as the
+   * number of memory we are going to create */
   g_return_val_if_fail (!wrapped_data
-      || views * n_mem != n_wrapped_pointers, FALSE);
+      || n_mem * views == n_wrapped_pointers, FALSE);
 
   for (v = 0; v < views; v++) {
     for (i = 0; i < n_mem; i++) {
-- 
2.11.0

